test/extensions/filters/network/thrift_proxy/conn_manager_test.cc

```diff
@@ -1925,6 +1936,127 @@ stat_prefix: test
   EXPECT_EQ(nullptr, route);
 }
 
+TEST_F(ThriftConnectionManagerTest, UnknownCluster) {
+  const std::string yaml = R"EOF(
+transport: FRAMED
+protocol: BINARY
+stat_prefix: test
+route_config:
+  name: "routes"
+  routes:
+    - match:
+        method_name: name
+      route:
+        cluster: cluster
+)EOF";
+
+  EXPECT_THROW_WITH_REGEX(initializeFilter(yaml), EnvoyException, "unknown thrift cluster");
+}
+
+TEST_F(ThriftConnectionManagerTest, UnknownCluster2) {
+  const std::string yaml = R"EOF(
+transport: FRAMED
+protocol: BINARY
+stat_prefix: test
+route_config:
+  name: "routes"
+  routes:
+    - match:
+        method_name: name
+      route:
+        cluster: cluster
+    - match:
+        method_name: name2
+      route:
+        cluster: unknown_cluster
+)EOF";
+
+  EXPECT_THROW_WITH_REGEX(initializeFilter(yaml, {"cluster"}), EnvoyException,
+                          "unknown thrift cluster");
+}
+
+TEST_F(ThriftConnectionManagerTest, UnknownCluster3) {
```

Comment:
Do we really need this test case?

Reply:
it's kinda random tests, but I could merge 2 and 3 to make it more elegant